flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -164,6 +172,8 @@ void compileFactory() {
 
             checkPatternSkipStrategy();
 
+            checkPatternWindowTimes();
```

Comment:
I guess this check should be moved after createStartState as windowTimes and windowTime are empty here.